src/entity2/revision.ts

```diff
@@ -0,0 +1,55 @@
+import { Entity, PrimaryGeneratedColumn, Column, BaseEntity, JoinColumn, ManyToOne } from 'typeorm';
+
+// eslint-disable-next-line import/no-cycle
+import { Dataset } from './dataset';
+import { User } from './user';
+
+interface Revision {
+    id: string;
+    revision_index: number;
+    dataset: Dataset;
+    creation_date: Date;
+    previous_revision: Revision;
+    online_cube_filename: string;
+    publish_date: Date;
+    approval_date: Date;
+    approved_by: User;
+    created_by: User;
+}
+
+@Entity()
+export class RevisionEntity extends BaseEntity implements Revision {
+    @PrimaryGeneratedColumn('uuid')
+    id: string;
+
+    @Column({ type: 'int' })
+    revision_index: number;
+
+    @ManyToOne(() => Dataset, (dataset) => dataset.revisions)
```

Comment:
ManyToOne?
For each revision there is only one dataset.
But for each dataset there can be many revisions.
From the POV of the `class RevisionEntity`, shouldn't it be OneToOne or perhaps OneToMany or have I got it backwards somehow? Afterall `(dataset) => dataset.revisions` could be reasonable but the syntax is bending my mind.

Reply:
Because the ownership is on the part of the Revision not the dataset... So this is the other side of that OneToMany relationship... e.g 1 Dataset, Many revisions.  Here in the revision entity we have only one dataset... but in the dataset entity we have an array of revisions.   Equally that revisions in the dataset are annotted with a `@OneToMany` annotation to match this `@ManyToOne`